### ISSUE TITLE
Run tests on automation with debug assertions enabled.

### ISSUE THREAD

**Issue by emilio** [servo/servo#13127]
Or at least a subset of them

**Comment by emilio** [comment]
cc @larsbergstrom @aneeshusa, could this be possible?

**Comment by emilio** [comment]
This is becoming an increasing priority as we implement parts of CSSOM, like https://github.com/servo/servo/pull/13176, where we **need** to check that the property declarations are accessed from where we think they're accessed, but that check can't be a release assertion (since it'd be basically a TLS lookup in extremely hot paths).

cc @SimonSapin @bholley

**Comment by SimonSapin** [comment]
#13177 discusses some of these `debug_assert!`

**Comment by larsbergstrom** [comment]
The only real issues that I see with running another configuration are:
1) cost (not a huge deal)
2) increased intermittents (a huge deal)

That said, I'm definitely open to it! If we did this, I'd also like to turn on integer overflow checking by default, too. That's something I'd love to see us *shipping*, and this would be a nice way to start it.

cc @edunham, too

**Comment by aneeshusa** [comment]
The `linux-dev` builder is already running [some tests](https://github.com/servo/saltfs/blob/654e20615a1b7220ab3983bdda2035663976a220/buildbot/master/files/config/steps.yml#L40); which tests specifically need to be added? Feel free to submit a PR to that file in saltfs.

If we're worried about intermittents, we may want to try it not-gating at first and see how bad it is before gating.

**Comment by emilio** [comment]
@aneehusa: Sorry, I missed that comment.

Yeah, I know we run unit tests and such in debug, but this is not comparable with wpt or similar.

Running WPT in debug mode was too slow IIRC, we should ideally make the release builder run with debug assertions.

**Comment by aneeshusa** [comment]
In that case, I think the easiest thing to do is change `mach` to add `-C debug-assertions=true` if an env var is set to `"1"` (say `SERVO_DEBUG_ASSERTIONS`), and set that environment variables to our [saltfs build environment](https://github.com/servo/saltfs/blob/master/buildbot/master/files/config/environments.py) (probably in `build_linux` for now). We can land the env var change first (it won't do anything), and then the `mach` change to see how bad it is in terms of intermittents.

An alternate is creating a new `linux-rel-debug-assertions` builder with the env var, and leaving the existing one untouched.

**Comment by aneeshusa** [comment]
The first PR to add the env var to saltfs is servo/saltfs#480.

**Comment by aneeshusa** [comment]
We'll need to wait for those changes to be rolled out (likely next week since today is a Read-Only Friday) before we can test the second PR changing `mach`.

**Comment by cyplo** [comment]
Hi ! Are the assertions enabled already, or do we need something else to happen to close this one ?

**Comment by jdm** [comment]
They were enabled. Thanks for the reminder!
